odbc/src/api.rs

```diff
@@ -709,16 +895,87 @@ pub extern "C" fn SQLGetDiagRec(
 
 #[no_mangle]
 pub extern "C" fn SQLGetDiagRecW(
```

Comment:
I've closed the comment where I was mentioning this. The error message returned should follow ODBC specific syntax. See https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/diagnostic-messages for more details on the format.
Right now, I don't see where _[ vendor-identifier ][ ODBC-component-identifier ]_ is set. What do you think makes more sense? _[ vendor-identifier ]_ should be the same everywhere and be something like _[MongDB Inc.]_ or _[MongoDB]_. For _[ODBC-component-identifier]_, it could be part of the ODBCError struct, and be something like _[api],_ _[core]_, _[driver]_ (driver here meaning the error came from the Rust driver we'll be using in the core api) for example. But it should still needs to be provided and added to the error message I think because I don't see how it can be set by the DM. Did you test this out as an integration tests to see? If not, maybe we can create a separate ticket to investigate/address this after this PR is merged? WDYT?

Reply:
I actually made those changes in the most recent commit -- I believe I've incorporated all of the formatting requirements. Do you think we still need integration tests?